### PR TITLE
Fold transition

### PR DESCRIPTION
**Pull request by tbaranes**
Here is the next one: Fold transition, related to #155 

That PR implements the `Fold` transition, it's working as expected, and can be customise with using params: `Fold(nbFolds)` which will change the number of pan during the transition, default is 2.

Missing:
- Reverse: I didn't find a way to detect if the transition is a push or a pop. Did you implement a system for that? The navigation is currently in hard coding (a private property set to true)
- Direction: not sure if that's needed, but after some tests I think it could be nice to setup the direction. Moreover, it could be used to solve the reverse issue if there no existing code to detect the that. 

The main code won't move (at least, if we dont implement the direction which is way more complex 😆) that much, so if you have any feedback, let me know 👐 

/!\ This PR is not finished yet, don't merge it /!\

### CODE REVIEW

**Comment by JakeLin** [comment]
@tbaranes thanks for the PR, great one 😎

Currently, the way to support reverse is to use direction eg. left vs right, in vs out. Because if we can support reverse transition effect, we should be able to expose the parameter, in `FoldAnimator` we can support `FoldAnimator(.Right, nbFolds)` and `FoldAnimator(.Left, nbFolds)`

 `FoldAnimator(.Right, nbFolds)`  will set `reverse = false` and  `FoldAnimator(.Right, nbFolds)`  will set `reverse = true`

Please let me know if you have any questions.

**Comment by tbaranes** [comment]
👍 Will work on it next week!

**Comment by tbaranes** [comment]
Here a small update:
- Small refactoring of the `AnimationTransitonType` to handle default direction, and direction + params
- The fold now handles direction to manage the `revere` animation

Still missing: vertical fold animations, it's only working with the horizontal one

Not sure about the buddy build failure, do I have something to do?

**Comment by tbaranes** [comment]
The vertical animation is here! Ready to merge from my side, what do you think about it @JakeLin ? 🎉

File: IBAnimatable/FoldAnimator.swift [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+  
+  func valuesForAxe(initialValue: CGFloat, reverseValue: CGFloat) -> (CGFloat, CGFloat) {
+    return horizontal ? (initialValue, reverseValue) : (reverseValue, initialValue)
+  }
+  
+  func width() -> CGFloat {
+    return horizontal ? size.width : size.height
+  }
+  
+  func height() -> CGFloat {
```
Comment (JakeLin) [inline IBAnimatable/FoldAnimator.swift]: can them be properties?

**Comment by JakeLin** [comment]
@tbaranes  well done, 👍 vertical animations fit most of the use cases. please get it in when you free ready. Thanks.

**Comment by tbaranes** [comment]
Thanks Jake! If travis build passed, I will merge it 👯

**Comment by JakeLin** [comment]
@tbaranes thanks again, I have fixed buddybuild by uploading the provisioning profile and p12 files. It is green again.

**Pull request merged**